Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -479,4 +479,164 @@ HRESULT _DWriteFontGetBoundingBoxesForGlyphs(
     });
 
     return ret;
-}
\ No newline at end of file
+}
+
+#pragma region CGFontCreateWithDataProvider helpers
+
+/**
+ * Custom implementation of IDWriteFontFileStream that implements Read in terms of an underlying CFDataRef
+ */
+class DWriteFontBinaryDataStream : public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IDWriteFontFileStream> {
+protected:
+    InspectableClass(L"Windows.Bridge.DirectWrite", TrustLevel::BaseTrust);
+
+public:
+    DWriteFontBinaryDataStream() {
+    }
+
+    HRESULT RuntimeClassInitialize(CGDataProviderRef dataProvider) {
+        _data.reset(CGDataProviderCopyData(dataProvider));
```

Comment:
? Taking a best guess that you want me to [dataProvider bytes]:
CGDataProvider is only a subclass of NSData on our platform.

We should not [CGDataProvider bytes] as it's taking advantage of an internal implementation detail that's subject to change.
We cannot use CGDataProviderCallbacks for access to underlying buffer, as we have not implemented those APIs.

Therefore we use the public interface, CGDataProviderCopyData(), which copies to a CGDataRef

---
In reply to: [88108446](https://github.com/Microsoft/WinObjC/pull/1386#discussion_r88108446) [](ancestors = 88108446)